src/web/mjs/connectors/MangaTown.mjs

```diff
@@ -56,23 +51,32 @@ export default class MangaTown extends Connector {
             };
         });
     }
-
     async _getPages(chapter) {
         let request = new Request(new URL(chapter.id, this.url), this.requestOptions);
         let data = await this.fetchDOM(request, 'div.manga_read_footer div.page_select select option');
-        return data
+        let imgpages = data
             .filter(option => !option.value.endsWith('featured.html'))
-            .map(element => this.createConnectorURI(this.getAbsolutePath(element.value, request.url)));
+            .map(element => this.getAbsolutePath(element.value, request.url));
+        //NOW we have all the pages, get the picture url from each page
+        //and create the payloads with mangahere referer
+        let imglist = [];
+        for (let i = 0; i<imgpages.length; i++ ) {
+            let request = new Request(imgpages[i], this.requestOptions);
+            data = await this.fetchDOM( request, 'source#image' );
```

Comment:
I strongly recommend to move the content of this loop as part into `_handleConnectorURI`.
The `_getPages` method is supposed to be fast and should not make too many requests if possible.
e.g.,

```javascript
async _handleConnectorURI(payload) {
    const pageData = await this.fetchDOM(new Request(payload, this.requestOptions), 'source#image');
    const imageURL = this.getAbsolutePath(data[0].src, request.url).replace('hakuneko://', 'https://');
    const request = new Request(imageURL, this.requestOptions);
    request.headers.set('x-referer', 'mangahere.com');
    const response = await fetch(request);
    const imageData = await response.blob();
    return this._blobToBuffer(imageData);
}
```

Reply:
Done